src/components/Table/Table.tsx

```diff
@@ -0,0 +1,110 @@
+import * as React from "react";
+import {
+  Table as ChakraTable,
+  Thead as ChakraTHead,
+  Tbody as ChakraTbody,
+  Tr as ChakraTr,
+  Th as ChakraTh,
+  Td as ChakraTd,
+  TableCaption as ChakraTableCaption,
+  useMultiStyleConfig,
+} from "@chakra-ui/react";
+
+import generateUUID from "../../helpers/generateUUID";
+
+export interface TableProps {
+  /** Additional class name for the `Table` component. */
+  className?: string;
+  /** Array of string values used to populate the `Table` column headers. */
+  columnHeaders?: string[];
+  /** Hex value to set the background color of the column headers. */
+  columnHeadersBackgroundColor?: string;
+  /** Hex value to set the text color of the column headers. */
+  columnHeadersTextColor?: string;
+  /** ID that other components can cross reference for accessibility purposes. */
+  id?: string;
+  /** If true, the first cell of each row in the `Table` component will be visually styled as a header.  The default value is false */
+  useRowHeaders?: boolean;
+  /** If true, a border will be displayed between each row in the `Table` component.  The default value is false. */
+  showRowDividers?: boolean;
+  /** Two-dimensional array used to populate the table rows. */
+  tableData: string[][];
+  /** Displays `Table` title element. */
+  titleText?: string;
+}
+
+function Table(props: React.PropsWithChildren<TableProps>) {
+  const {
+    className,
+    columnHeaders,
+    columnHeadersBackgroundColor,
+    columnHeadersTextColor,
+    id = generateUUID(),
+    useRowHeaders = false,
+    showRowDividers = false,
+    tableData,
+    titleText,
+  } = props;
+
+  const customColors = {};
+
+  columnHeadersBackgroundColor &&
+    (customColors["backgroundColor"] = columnHeadersBackgroundColor);
+  columnHeadersTextColor && (customColors["color"] = columnHeadersTextColor);
+
+  const styles = useMultiStyleConfig("CustomTable", {
+    showRowDividers,
+    useRowHeaders,
+    columnHeadersTextColor,
+  });
+
+  const columnHeadersElems = columnHeaders && (
+    <ChakraTHead>
+      <ChakraTr>
+        {columnHeaders.map((child, key) => (
+          <ChakraTh key={key} sx={customColors}>
+            {child}
+          </ChakraTh>
+        ))}
+      </ChakraTr>
+    </ChakraTHead>
+  );
+
+  const tableBodyElems = () => {
+    /** tableData value should be two dimensional array */
+    if (
+      !Array.isArray(tableData) ||
+      tableData.length <= 0 ||
+      tableData[0].constructor !== Array
+    ) {
+      console.warn(`Table data should be two dimensional array.`);
+      return null;
+    }
+
+    return (
+      <ChakraTbody>
+        {tableData.map((row, index) => (
+          <ChakraTr key={index}>
+            {row.map((column, key) => (
+              <ChakraTd key={key}>{column}</ChakraTd>
+            ))}
+          </ChakraTr>
```

Comment:
When `useRowHeaders` is true, besides making the first cell bold, it should be a `tr` instead of a `td`. According to the spec document, it also needs a `scope="row"`, 
`Each header cell must have an appropriate scope attribute. Either scope="col" or scope="row"`".

So, for example, when `useRowHeaders` is true, it should render something like
```
<tr>
  <th scope="row">first</th>
  <td>second</td>
  <td>third</td>
</tr>
```

This this as an example: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/th